virttest/utils_test/libvirt.py

```diff
@@ -3550,3 +3550,25 @@ def check_qemu_cmd_line(content, err_ignore=False):
             raise exceptions.TestFail("Expected '%s' was not found in "
                                       "qemu command line" % content)
     return True
+
+
+def get_disk_alias(vm, source_file=None):
+    """
+    Get alias name of disk with given source file
+
+    :param vm: VM object
+    :param source_file: domain disk source file
+    """
+    if vm.is_alive():
+        ori_vmxml = vm_xml.VMXML.new_from_dumpxml(vm.name)
+        disks = ori_vmxml.devices.by_device_tag('disk')
+        for disk in disks:
+            if disk.xmltreefile.find('source') is not None:
+                if disk.source.attrs['file'] == source_file:
+                    logging.info("Get alias name %s", disk.alias['name'])
+                    return disk.alias['name']
```

Comment:
line 3569 and 3573 return the same :disk.alias['name'] , so may not diff them

Reply:
there are 2 scenario I need to catch, first is find the disk according to source path, second is to find the no source disk, I don't know how to merge their return here

Reply:
yes, same concern here.

Reply:
@hahakiki2010 can you give me some solution for it? I don't know more clear for your concern

Reply:
The concern is if there are multiple disks, some of them have source file , some of them don't, then the code can't assure to return the disk having the given source file. Maybe the logic could be:
'''
for disk in disks:
  if disk.xmltreefile.find('source') is not None and 'file' in disk.source.attrs:
      return disk.alias['name']
if source_file is None:
  for disk in disks:
    # Just return the first disk alias name
    return disk.alias['name']
return None
'''

Reply:
@hahakiki2010 Then if you want to get the disk alias with source, then give the source path, if not, will find the no source disk. I set the judgement "disk.xmltreefile.find('source') is not None" in the first level just to make sure it have source or not.